pkg/gencosmosdb/cosmosdb/template.go

```diff
@@ -145,7 +157,7 @@ func (c *templateClient) setOptions(options *Options, template *pkg.Template, he
 	}
 
 	if !options.NoETag {
-		if template.ETag == "" {
+		if template != nil && template.ETag == "" {
```

Comment:
this doesn't make sense because you would still dereference a nil template below in `template.Etag`.  Perhaps you want `if template != nil && !options.NoETag {` above?